### PR TITLE
Feature/config type script

### PR DESCRIPTION
**Pull request by vinhlee95**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,15 +11,24 @@
   "author": "",
   "license": "ISC",
   "devDependencies": {
+    "awesome-typescript-loader": "^5.2.1",
```
Comment (HuyAms) [inline package.json]: Babel7 is now supporting TS, no need awesome-typescript-loader. 
Add "@babel/typescript" to .babelrc

File: package.json [hunk 2]
```diff
@@ -11,15 +11,24 @@
...
     "clean-webpack-plugin": "^2.0.1",
     "css-loader": "^2.1.1",
     "html-webpack-plugin": "^3.2.0",
+    "ts-loader": "^5.3.3",
```
Comment (HuyAms) [inline package.json]: Babel7 is now supporting TS, no need ts-loader

File: webpack/webpack.common.js [hunk 3]
```diff
@@ -3,72 +3,80 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
...
-  plugins: [
-    new HtmlWebpackPlugin({
-      template: path.resolve(__dirname, '../public/index.html')
-    })
-  ]
+	module: {
+		rules: [
+			{
+				test: /\.tsx?$/,
+				loader: 'awesome-typescript-loader'
```
Comment (HuyAms) [inline webpack/webpack.common.js]: add babel-loader instead

File: webpack/webpack.common.js [hunk 4]
```diff
@@ -3,72 +3,80 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
...
+					}
+				]
+			}
+		]
+	},
+
+	plugins: [
+		new HtmlWebpackPlugin({
+			template: path.resolve(__dirname, '../public/index.html')
+		})
```
Comment (HuyAms) [inline webpack/webpack.common.js]: add ForkTsCheckerWebpackPlugin for type-check

File: webpack/webpack.common.js [hunk 5]
```diff
@@ -1,81 +1,91 @@
...
+							pngquant: {
+								quality: '65-90',
+								speed: 4
+							}
+						}
+					}
+				]
+			}
+		]
+	},
```
Comment (vinhlee95) [inline webpack/webpack.common.js]: This is freaking weird spacing issues, don't know if prettier could fix this but let's deal w it later

**Pull request merged**